libcxx/include/__atomic/atomic_base.h

```diff
@@ -131,6 +154,11 @@ struct __atomic_base // false
   __atomic_base(const __atomic_base&) = delete;
 };
 
+#if _LIBCPP_STD_VER >= 20
+#  undef _LIBCPP_REQUIRE_IS_ALWAYS_LOCK_FREE
+#  define _LIBCPP_REQUIRE_IS_ALWAYS_LOCK_FREE requires __base::is_always_lock_free
```

Comment:
I really dislike this, it makes it very hard to determine what the macro `_LIBCPP_REQUIRE_IS_ALWAYS_LOCK_FREE` is defined to. 

In the `__config` header we have some macros to conditionally do things, for example `__NOEXCEPT_(...)`. There is no `_LIBCPP_REQUIRES(...)` since there was no need for that macro. Can you add that macro for this patch?

Reply:
You're right, that is a better idea. I will make that change.

Since this is gated to versions >= C++20, perhaps I should add a version gate as well e.g.  `_LIBCPP_REQUIRES_IN_CXX20`?